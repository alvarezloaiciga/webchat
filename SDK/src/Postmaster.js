// @flow
import postRobot from 'post-robot/dist/post-robot.ie';
import {eventTypes, bridgePath} from 'Common/Constants';
import {
  displayError,
  displayWarning,
  isIFrame,
  getBrowserName,
  isStorageEnabled,
  isSupportedBrowser,
} from 'Common/Utils';
import {getChatWindow, getQuiqOptions} from './Globals';

postRobot.CONFIG.LOG_LEVEL = 'error';

const handlers: {[string]: Array<?(data: Object) => any>} = {};
let postRobotClient, postRobotListener;
let listeners = [];

/*****************************************************
 * ABOUT EVENT HANDLING
 * We listen for each supported event type on the postRobot messaging channel between the SDK and the webchat iframe
 * Each event type is associated with an array of event handling functions. Anything may register a function to handle an event.
 * For example, Quiq.js will register an eventHandler when the user calls the .on() function.
 * When an event is triggered, event handler functions are called in the order they appear in the array.
 *****************************************************/

export const setup = () => {
  const chatWindow = getChatWindow();
  const {host, clientDomain} = getQuiqOptions();

  cancelListeners();

  // Because of an oddity in post-robot, we can't pass an iframe into postRobot.listener(). Need to find the associated window.
  const targetWindow = isIFrame(chatWindow) ? chatWindow.contentWindow : chatWindow;

  try {
    // Build cross-domain bridge (for IE compatibility)
    if ((getBrowserName() === 'IE' || getBrowserName() === 'Edge') && clientDomain !== host) {
      postRobot.bridge.openBridge(`${host}/${bridgePath}`);
    }
  } catch (e) {
    /* eslint-disable no-console */
    console.warn(`Error building postRobot Bridge: ${e.message}\nProceeding as normal.`);
    /* eslint-disable no-console */
  }

  postRobotClient = postRobot.client({window: targetWindow, timeout: 2000, host});
  postRobotListener = postRobot.listener({window: targetWindow, host});
  setupListeners();
};

export const registerEventHandler = (event: string, handler: (data: Object) => any) => {
  // Ensure this event has a key in the handlers object, if not create it and assign an empty array to it
  if (!Array.isArray(handlers[event])) {
    handlers[event] = [];
  }
  handlers[event].push(handler);
};

export const removeEventHandler = (event: string, handler: (data: Object) => any) => {
  if (Array.isArray(handlers[event])) {
    const idx = handlers[event].indexOf(handler);
    if (idx > -1) {
      handlers[event].splice(idx, 1);
    }
  }
};

export const tellChat = (messageName: string, data: Object = {}) => {
  if (!isStorageEnabled() || !isSupportedBrowser()) {
    displayError('Client browser did not meet all requirements for SDK communication. Aborting');
  }

  if (!postRobotClient) {
    displayError(
      'You must set the webchat window and domain, and then call Postmaster.setup(), before trying to post a message!',
    );
  }
  postRobotClient.send(messageName, data);
};

export const askChat = async (
  messageName: string,
  data: ?Object,
  callback: ?(data: ?Object, error: ?Error) => any,
): Promise<Object> => {
  if (!isStorageEnabled() || !isSupportedBrowser()) {
    displayError('Client browser did not meet all requirements for SDK communication. Aborting');
  }

  if (!postRobotClient) {
    displayError(
      'You must set the webchat window and domain, and then call Postmaster.setup(), before trying to post a message!',
    );
  }

  try {
    const response = await postRobotClient.send(messageName, data || {});

    if (callback) {
      callback(response.data, null);
    }

    return response.data;
  } catch (error) {
    if (callback) callback(null, error);

    throw error;
  }
};

const setupListeners = () => {
  if (!postRobotListener) {
    displayError(
      'You must set the webchat window and domain, and then call Postmaster.setup(), before trying to create a message listener',
    );
  }

  listeners = Object.values(eventTypes).map(e => {
    // $FlowIssue - flow cannot infer type of Object.{entries, keys, values}
    return postRobotListener.on(e, handleEvent(e));
  });
};

const cancelListeners = () => {
  listeners.forEach(listener => listener.cancel());
  listeners = [];
};

const handleEvent = (eventName: string) => event => {
  if (handlers.hasOwnProperty(eventName)) {
    handlers[eventName].forEach(f => f && f(event.data));
  } else if (!Object.values(eventTypes).includes(eventName)) {
    displayWarning(`QUIQ SDK: Unknown event type received: ${eventName}`);
  }
};
