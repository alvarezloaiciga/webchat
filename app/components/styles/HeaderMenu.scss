$HeaderMenuColor: #fff;

.HeaderMenu {
  display: flex;
  justify-content: flex-end;
  padding: 5px 7px 3px;
  font-weight: 300;
  min-height: 25px;
  flex: 0 0 auto;
  font-size: 14px;
  color: $HeaderMenuColor;
  border-bottom: 1px solid white;

  .icon {
    stroke: white;
    margin-left: 7px;
    cursor: pointer;
    border-bottom: 1px solid transparent;

    /* Account for the different size of fa-times close icon */
    &.fa-times {
      margin-bottom: 2px;
      font-size: 20px;
      line-height: 14px;
    }

    &:hover {
      color: darken($HeaderMenuColor, 10%);
      border-bottom: 1px solid white;
    }
  }

  .buttons {
    display: flex;
    justify-content: center;
    align-items: center;
  }
}

.noFlexbox .HeaderMenu {
  height: 20px;
  position: absolute;
  top: 0;
  left: 0;
  width: 388px;
  max-width: 388px;
  min-width: 388px;
  min-height: 20px;
  max-height: 20px;

  .buttons {
    float: right;
  }
}
