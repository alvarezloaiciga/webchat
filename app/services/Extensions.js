// @flow

export type ExtensionEvent = {
  eventType: string,
  data: Object,
};

let _extensionWindow: Object;
let _extensionUrl: string;

export const registerExtension = (extensionUrl: string, extensionWindow: Object) => {
  _extensionUrl = extensionUrl;
  _extensionWindow = extensionWindow;
};

export const postExtensionEvent = (event: ExtensionEvent) => {
  if (_extensionWindow) {
    _extensionWindow.postMessage(event, _extensionUrl);
  }
};
