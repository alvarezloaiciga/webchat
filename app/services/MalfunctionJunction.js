// @flow
/**********************************************************************************
 Malfunction Junction is a dispatch service, designed to be paired with our Quiq.js
 SDK. It listens for messages sent via a postRobot channel, and then dispatches
 calls to Redux actions or the QuiqChatClient singleton.

 Before use, an instance of QuiqChatClient must be registered,
 as well as the default Redux store.
 **********************************************************************************/

import postRobot from 'post-robot/dist/post-robot.ie';
import watchStore from 'redux-store-watch';
import * as ChatActions from 'actions/chatActions';
import * as ChatSelectors from 'reducers/chat';
import {eventTypes, actionTypes} from 'Common/Constants';
import {displayError, getHostingWindow, getWindowDomain} from 'Common/Utils';
import messages from 'messages';
import type {ReduxStore, ChatState} from 'types';

let reduxWatch;
let dispatch;
let getState: () => ChatState;
let chatClient;
let domain;
let postRobotClient, postRobotListener;

export const init = (_domain: string, _store: ReduxStore, _chatClient) => {
  dispatch = _store.dispatch;
  getState = _store.getState;
  chatClient = _chatClient;
  domain = _domain;

  reduxWatch = watchStore(_store);

  // Listen to parent/host page
  const hostWindow = getHostingWindow();
  postRobotClient = postRobot.client({window: hostWindow, domain});
  postRobotListener = postRobot.listener({window: hostWindow, domain});

  setupListeners();
  setupReduxHooks();
};

const setupListeners = () => {
  if (!postRobotListener) {
    return displayError(messages.mfInitNeeded);
  }

  postRobotListener.on(actionTypes.setChatVisibility, setChatVisibility);
  postRobotListener.on(actionTypes.getChatVisibility, getChatVisibility);
  postRobotListener.on(actionTypes.getAgentAvailability, getAgentAvailability);
};

const tellClient = (messageName: string, data: Object = {}) => {
  if (!postRobotClient) {
    return displayError(messages.mfInitNeeded);
  }

  postRobotClient.send(messageName, data);
};

/**********************************************************************************
 * "Automatic event hooks (webchat -> client)
 **********************************************************************************/
const setupReduxHooks = () => {
  if (!reduxWatch) {
    return displayError(messages.mfInitNeeded);
  }

  // Send events to SDK in response to changes in redux state
  // ChatContainer visibility
  reduxWatch.watch(ChatSelectors.getChatContainerHidden, hidden =>
    tellClient(eventTypes.chatVisibilityDidChange, {visible: !hidden}),
  );

  // Agent availability
  reduxWatch.watch(ChatSelectors.getAgentsAvailable, available =>
    tellClient(eventTypes.agentAvailabilityDidChange, {available}),
  );
};

/**********************************************************************************
 * Manual event hooks (webchat -> client)
 **********************************************************************************/

export const chatVisibilityDidChange = (visible: boolean) => {
  tellClient(eventTypes.chatVisibilityDidChange, {visible});
};

export const standaloneOpen = () => {
  tellClient(eventTypes.standaloneOpen);
};

/**********************************************************************************
 * View facade handlers (client -> webchat)
 **********************************************************************************/

const setChatVisibility = (event: Object) => {
  const {visible} = event.data;
  dispatch(ChatActions.setChatContainerHidden(!visible));
};

const getChatVisibility = () => {
  return {visible: !ChatSelectors.getChatContainerHidden(getState())};
};

const getAgentAvailability = () => {
  return {available: ChatSelectors.getAgentsAvailable(getState())};
};

/**********************************************************************************
 * Controller facade handlers (client -> webchat)
 **********************************************************************************/

// TODO
