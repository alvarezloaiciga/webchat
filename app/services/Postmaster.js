// @flow
/**********************************************************************************
 Malfunction Junction is a dispatch service, designed to be paired with our Quiq.js
 SDK. It listens for messages sent via a postRobot channel, and then dispatches
 calls to Redux actions or the QuiqChatClient singleton.

 Before use, an instance of QuiqChatClient must be registered,
 as well as the default Redux store.
 **********************************************************************************/

import postRobot from 'post-robot/dist/post-robot.ie';
import watchStore from 'redux-store-watch';
import * as ChatActions from 'actions/chatActions';
import * as ChatSelectors from 'reducers/chat';
import {eventTypes, actionTypes} from 'Common/Constants';
import {displayError, getHostingWindow, getQuiqKeysFromLocalStorage} from 'Common/Utils';
import type {RegistrationField} from 'Common/types';
import {constructApp, appIsMounted} from 'utils/domUtils';
import QuiqChatClient from 'quiq-chat';
import quiqOptions from 'Common/QuiqOptions';
import type {ReduxStore} from 'types';

let reduxWatch;
let store;
let domain;
let postRobotClient, postRobotListener;

postRobot.CONFIG.LOG_LEVEL = 'error';

export const init = (_domain: string, _store: ReduxStore) => {
  store = _store;
  domain = _domain;

  reduxWatch = watchStore(_store);

  // Listen to parent/host page
  const hostWindow = getHostingWindow();
  postRobotClient = postRobot.client({window: hostWindow, domain});
  postRobotListener = postRobot.listener({window: hostWindow, domain});

  setupListeners();
  setupReduxHooks();
};

const setupListeners = () => {
  if (!postRobotListener) {
    return displayError('Postmaster.init() must be called prior to setting up listeners.');
  }

  postRobotListener.on(actionTypes.setChatVisibility, setChatVisibility);
  postRobotListener.on(actionTypes.getChatVisibility, getChatVisibility);
  postRobotListener.on(actionTypes.getAgentAvailability, getAgentAvailability);
  postRobotListener.on(actionTypes.sendRegistration, sendRegistration);
};

export const tellClient = (messageName: string, data: Object = {}) => {
  if (!postRobotClient) {
    return displayError('Postmaster.init() must be called prior to sending message.');
  }

  postRobotClient.send(messageName, data);
};

/**********************************************************************************
 * "Automatic event hooks (webchat -> client)
 **********************************************************************************/
const setupReduxHooks = () => {
  if (!reduxWatch) {
    return displayError('Postmaster.init() must be called prior to sending message.');
  }

  // Send events to SDK in response to changes in redux state
  // ChatContainer visibility
  reduxWatch.watch(ChatSelectors.getChatContainerHidden, hidden =>
    tellClient(eventTypes.chatVisibilityDidChange, {visible: !hidden}),
  );

  // Chat launcher visibility
  reduxWatch.watch(ChatSelectors.getChatLauncherHidden, hidden =>
    tellClient(eventTypes._launchButtonVisibilityShouldChange, {
      visible: !hidden,
    }),
  );

  // Agent availability
  reduxWatch.watch(ChatSelectors.getAgentsAvailable, available =>
    tellClient(eventTypes.agentAvailabilityDidChange, {available}),
  );
};

/**********************************************************************************
 * Manual event hooks (webchat -> client)
 **********************************************************************************/

export const chatVisibilityDidChange = (visible: boolean) => {
  tellClient(eventTypes.chatVisibilityDidChange, {visible});
};

export const standaloneOpen = () => {
  store.dispatch(ChatActions.setChatContainerHidden(true));
  tellClient(eventTypes._standaloneOpen, {
    localStorageKeys: getQuiqKeysFromLocalStorage(quiqOptions.contactPoint),
  });
};

/**********************************************************************************
 * View facade handlers (client -> webchat)
 **********************************************************************************/

const setChatVisibility = (event: Object) => {
  const {visible} = event.data;

  if (visible && !appIsMounted()) {
    constructApp(store);
  }

  store.dispatch(ChatActions.setChatContainerHidden(!visible));
};

const getChatVisibility = () => {
  return {visible: !ChatSelectors.getChatContainerHidden(store.getState())};
};

/**********************************************************************************
 * Controller (QuiqChatClient) facade handlers (client -> webchat)
 **********************************************************************************/

const sendRegistration = (event: Object) => {
  const registrationData: Array<RegistrationField> = event.data.registrationData;
  const registrationDictionary: {[string]: string} = {};

  for (let index = 0; index < registrationData.length; index++) {
    const data = registrationData[index];
    registrationDictionary[data.id] = data.value;
  }

  QuiqChatClient.sendRegistration(registrationDictionary);
};

const getAgentAvailability = async () => await QuiqChatClient.checkForAgents();
