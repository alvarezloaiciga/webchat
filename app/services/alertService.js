import assets from 'assets';

// Load alert sound--must be done here, not inside alert function
// $FlowIssue
const canPlayMp3 = ['probably', 'maybe'].includes(new Audio().canPlayType('audio/mp3'));
const alertFile = canPlayMp3 ? assets.alertSound : assets.alertSoundWav;

// $FlowIssue
const alertSound = new Audio(alertFile);

/**
 * Plays a sound notification. Designed for demoing sound to user.
 */
export const playSound = () => {
  alertSound.play();
};
