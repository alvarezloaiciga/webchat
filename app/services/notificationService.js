// @flow

import {registerEventHandler} from 'Postmaster';
import {eventTypes} from 'Common/Constants';
import {getDisplayString} from 'Common/i18n';
import {getQuiqOptions} from 'Globals';

const handleAgentMessageArrived = () => {
  const options = getQuiqOptions();

  if (options.flashNotificationOnNewMessage) {
    flashTitle(getDisplayString(options.messages.messageArrivedNotification));
  }
};

export const init = () => {
  registerEventHandler(eventTypes.agentMessageArrived, handleAgentMessageArrived);
};

/**
 * Gets whether or not the app is visible to the user
 */
const appIsHidden = (): boolean => {
  const prefixes = ['webkit', 'moz', 'ms', 'o'];

  if ('hidden' in document) {
    return document.hidden;
  }

  for (let i = 0; i < prefixes.length; i++) {
    if (`${prefixes[i]}Hidden` in document) {
      // $FlowIssue
      return document[`${prefixes[i]}Hidden`];
    }
  }

  // At this point it's not supported, returning false as default
  return false;
};

let oldTitle;

/**
 * Handles visibility change event, resetting title and removing related timers if app is visible again
 */
const appDidBecomeVisible = () => {
  if (appIsHidden()) return;

  // Reset title and flash timers
  window.clearInterval(window.titleFlashTimer);
  document.title = oldTitle;
  document.removeEventListener('visibilitychange', appDidBecomeVisible);
};

/**
 * Reset title and remove flash timers, regardless of visibility
 */
export const resetTitle = () => {
  // Reset title and flash timers
  window.clearInterval(window.titleFlashTimer);
  document.title = oldTitle;
  document.removeEventListener('visibilitychange', appDidBecomeVisible);
};

/**
 * Flashes the title bar if the app is blurred. Ceases when focus returns.
 * @param {string}  title           - The title to flash
 */
export const flashTitle = (title: string) => {
  if (!appIsHidden()) return;

  // Clear any existing flash timers
  if (window.titleFlashTimer) {
    window.clearInterval(window.titleFlashTimer);
  }

  oldTitle = document.title;
  window.titleFlashTimer = window.setInterval(function() {
    document.title = document.title === oldTitle ? title : oldTitle;
  }, 1000);

  // Supported in all browsers including IE >= 10. If we want to support Android Browser 4.4 we'll need a prefix
  document.addEventListener('visibilitychange', appDidBecomeVisible);
};
