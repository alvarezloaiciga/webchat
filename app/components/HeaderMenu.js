// @flow
import React, {Component} from 'react';
import {inStandaloneMode} from 'Common/Utils';
import {getStyle} from 'Common/QuiqOptions';
import {destructApp} from 'utils/domUtils';
import {intlMessageTypes, ChatInitializedState, displayModes} from 'Common/Constants';
import {setChatContainerHidden} from 'actions/chatActions';
import {connect} from 'react-redux';
import {standaloneOpen} from 'services/Postmaster';
import {getMessage, getConfiguration} from 'reducers/chat';
import type {ChatState, ChatInitializedStateType, ChatConfiguration} from 'types';
import './styles/HeaderMenu.scss';

export type HeaderMenuProps = {
  initializedState: ChatInitializedStateType,
  setChatContainerHidden: (chatContainerHidden: boolean) => void, // eslint-disable-line react/no-unused-prop-types
  configuration: ChatConfiguration,
};

export type HeaderMenuState = {
  openingWindowExists?: boolean,
};

export class HeaderMenu extends Component<HeaderMenuProps, HeaderMenuState> {
  props: HeaderMenuProps;
  state: HeaderMenuState = {
    openingWindowExists: true,
  };
  parentWindowExistenceCheckInterval: ?number;

  componentDidMount() {
    if (inStandaloneMode()) {
      this.parentWindowExistenceCheckInterval = setInterval(this.updateParentWindowExistence, 100);
    }
  }

  componentWillUnmount() {
    if (this.parentWindowExistenceCheckInterval) {
      clearInterval(this.parentWindowExistenceCheckInterval);
      this.parentWindowExistenceCheckInterval = null;
    }
  }

  minimize = () => {
    this.props.setChatContainerHidden(true);
  };

  popChat = () => {
    standaloneOpen();
    destructApp();
  };

  updateParentWindowExistence = () => {
    if (!window.opener || window.opener.closed) {
      this.setState({openingWindowExists: false});
      if (this.parentWindowExistenceCheckInterval) {
        clearInterval(this.parentWindowExistenceCheckInterval);
        this.parentWindowExistenceCheckInterval = null;
      }
    }
  };

  renderButtons = () => (
    <div className="buttons">
      {(!inStandaloneMode() || this.state.openingWindowExists) &&
        this.props.configuration.displayMode !== displayModes.UNDOCKED && (
          <i
            className={`fa fa-window-minimize icon`}
            title={getMessage(intlMessageTypes.minimizeWindowTooltip)}
            onClick={inStandaloneMode() ? window.close : this.minimize}
          />
        )}

      {this.props.initializedState !== ChatInitializedState.BURNED &&
        (!inStandaloneMode() || this.state.openingWindowExists) &&
        this.props.configuration.displayMode === displayModes.EITHER && (
          <i
            className={`fa fa-${inStandaloneMode() ? 'window-restore' : 'window-maximize'} icon`}
            title={getMessage(
              inStandaloneMode()
                ? intlMessageTypes.dockWindowTooltip
                : intlMessageTypes.openInNewWindowTooltip,
            )}
            onClick={inStandaloneMode() ? window.close : this.popChat}
          />
        )}
      <i
        className={`fa fa-times icon`}
        title={getMessage(intlMessageTypes.closeWindowTooltip)}
        onClick={inStandaloneMode() ? window.close : this.minimize}
      />
    </div>
  );

  render() {
    const {colors, styles, fontFamily} = this.props.configuration;

    const headerStyle = getStyle(styles.HeaderMenu, {backgroundColor: colors.primary});
    const titleTextStyle = getStyle(styles.TitleText, {fontFamily});

    return (
      <div className="HeaderMenu" style={headerStyle}>
        <div className="title">
          <span style={titleTextStyle}>{getMessage(intlMessageTypes.titleText)}</span>
        </div>
        {!this.props.configuration.demoMode && this.renderButtons()}
      </div>
    );
  }
}

export default connect(
  (state: ChatState) => ({
    initializedState: state.initializedState,
    configuration: getConfiguration(state),
  }),
  {setChatContainerHidden},
)(HeaderMenu);
