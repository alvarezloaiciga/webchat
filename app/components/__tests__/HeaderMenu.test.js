// @flow
jest.mock('quiq-chat');
jest.mock('Common/QuiqOptions');
jest.mock('services/Postmaster');

import React from 'react';
import type {HeaderMenuProps} from '../HeaderMenu';
import {HeaderMenu} from '../HeaderMenu';
import QuiqChatClient from 'quiq-chat';
import {shallow} from 'enzyme';
import {standaloneOpen} from 'services/Postmaster';
import type {ShallowWrapper} from 'enzyme';

describe('HeaderMenu component', () => {
  let wrapper: ShallowWrapper;
  let testProps: HeaderMenuProps;
  let render: () => void;

  beforeEach(() => {
    testProps = {
      setChatContainerHidden: jest.fn(),
      setChatPopped: jest.fn(),
      initializedState: 'initialized',
    };

    render = () => {
      wrapper = shallow(<HeaderMenu {...testProps} />);
    };
  });

  describe('rendering', () => {
    it('renders', () => {
      render();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('minimize', () => {
    it('calls minimize', () => {
      render();
      wrapper.find('.fa-window-minimize').simulate('click');
      expect(testProps.setChatContainerHidden).toBeCalledWith(true);
    });
  });

  describe('when burned', () => {
    it('hides maximize', () => {
      testProps.initializedState = 'burned';
      render();
      expect(wrapper.find('.fa-window-restore').length).toBe(0);
    });
  });

  // TODO: Re-enable me once iFrame is merged in.
  /* eslint-disable no-restricted-syntax */
  xdescribe('popChat', () => {
    it('pops chat', () => {
      render();
      wrapper.find('.fa-window-maximize').simulate('click');
      expect(standaloneOpen).toBeCalled();
    });
  });
  /* eslint-disable no-restricted-syntax */
});
