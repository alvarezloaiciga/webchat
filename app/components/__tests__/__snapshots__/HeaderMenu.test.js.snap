// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenu component rendering renders 1`] = `
<div
  className="HeaderMenu"
  style={
    Object {
      "background": "#31bf8b",
      "backgroundColor": "#333",
    }
  }
>
  <div
    className="title"
  >
    <span
      style={
        Object {
          "fontFamily": "Lato, sans-serif",
        }
      }
    />
  </div>
  <div
    className="buttons"
  >
    <i
      className="fa fa-window-minimize icon"
      onClick={[Function]}
      title="Minimize window"
    />
    <i
      className="fa fa-window-maximize icon"
      onClick={[Function]}
      title="Open chat in new window"
    />
    <i
      className="fa fa-times icon"
      onClick={[Function]}
      title="Close window"
    />
  </div>
</div>
`;

exports[`HeaderMenu component rendering when not in standalone mode renders 1`] = `
<div
  className="HeaderMenu"
  style={
    Object {
      "background": "#31bf8b",
      "backgroundColor": "#333",
    }
  }
>
  <div
    className="title"
  >
    <span
      style={
        Object {
          "fontFamily": "Lato, sans-serif",
        }
      }
    />
  </div>
  <div
    className="buttons"
  >
    <i
      className="fa fa-window-minimize icon"
      onClick={[Function]}
      title="Minimize window"
    />
    <i
      className="fa fa-window-maximize icon"
      onClick={[Function]}
      title="Open chat in new window"
    />
    <i
      className="fa fa-times icon"
      onClick={[Function]}
      title="Close window"
    />
  </div>
</div>
`;
